src/google/protobuf/io/tokenizer.cc

```diff
@@ -413,8 +419,11 @@ void Tokenizer::ConsumeString(char delimiter) {
       }
 
       default: {
-        if (current_char_ == delimiter) {
-          NextChar();
+        if (delimiter == 0) {
+          if (TryConsume('"') && TryConsume('"') && TryConsume('"')) {
```

Comment:
This seems wrong. If there is only one quote, that should also terminate a triple-quoted block and cause its contents to be reinterpreted as a regular quoted literal (since the first three quotes means it was an empty string concatenated with a normal string). So there should be some different action taken if the first `TryConsume` succeeds but the second fails.

Similarly, if there are two quotes (instead of three), that should be interpreted as ending regular quoted literal and then starting a new regular quoted literal (whose value will be concatenated to the previous one).

But this code just falls through if it only consumes one or two quotes, which means it calls `NextChar()` on line 429. I think that means it eats/loses a character of input.

Reply:
You are correct and, iiuc, alluded to this in an earlier comment.  Limiting multi-line strings to backticks should solve this.